src/Http/Http.Extensions/src/UriHelper.cs

```diff
@@ -58,24 +61,29 @@ public static class UriHelper
                 throw new ArgumentNullException(nameof(scheme));
             }
 
-            var combinedPath = (pathBase.HasValue || path.HasValue) ? (pathBase + path).ToString() : "/";
```

Comment:
By removing this combine you're bypassing the slash deduplication logic from here:
https://github.com/dotnet/aspnetcore/blob/277cc564ce5a76c6613f29dd92c0168e06b41b1c/src/Http/Http.Abstractions/src/PathString.cs#L305-L312

If there's not already a test for that, please add one.

Reply:
Fixed!

Added test cases.